docs/csharp/linq/get-started/walkthrough-writing-queries-linq.md

```diff
@@ -0,0 +1,137 @@
+
+- The `Student` record consists of autoimplemented properties.
+- Each student in the list is initialized with the primary constructor.
+- The sequence of scores for each student is initialized with a primary constructor.
+
+Next, create a sequence of `Student` records that serves as the source of this query. Open *Program.cs*, and remove the following boilerplate code:
+
+```csharp
+// See https://aka.ms/new-console-template for more information
+Console.WriteLine("Hello, World!");
+```
+
+Replace it with the following code that creates a sequence of `Student` records:
+
+:::code language="csharp" source="./snippets/WalkthroughWritingLinqQueries/Program.cs" id="CreateSequence":::
+
+- The sequence of students is initialized with a collection expression.
+- The `Student` record type holds the static list of all students.
+- Some of the constructor calls use [named arguments](../../programming-guide/classes-and-structs/named-and-optional-arguments.md) to clarify which argument matches which constructor parameter.
+
+Try adding a few more students with different test scores to the list of students to get more familiar with the code so far.
+
+## Create the query
+
+Next, you create your first query. Your query, when you execute it, produces a list of all students whose score on the first test was greater than 90. Because the whole `Student` object is selected, the type of the query is `IEnumerable<Student>`. Although the code could also use implicit typing by using the [var](../../language-reference/statements/declarations.md#implicitly-typed-local-variables) keyword, explicit typing is used to clearly illustrate results. (For more information about `var`, see [Implicitly Typed Local Variables](../../programming-guide/classes-and-structs/implicitly-typed-local-variables.md).) Add the following code to *Program.cs*, after the code that creates the sequence of students:
+
+:::code language="csharp" source="./snippets/WalkthroughWritingLinqQueries/Program.cs" id="DefineFirstQuery":::
+
+The query's range variable, `student`, serves as a reference to each `Student` in the source, providing member access for each object.
+
+## Run the query
+
+Now write the `foreach` loop that causes the query to execute. Each element in the returned sequence is accessed through the iteration variable in the `foreach` loop. The type of this variable is `Student`, and the type of the query variable is compatible, `IEnumerable<Student>`. After you added the following code, build and run the application to see the results in the **Console** window.
+
+:::code language="csharp" source="./snippets/WalkthroughWritingLinqQueries/Program.cs" id="RunFirstQuery":::
+
+You can combine multiple Boolean conditions in the `where` clause in order to further refine a query. The following code adds a condition so that the query returns those students whose first score was over 90 and whose last score was less than 80. The `where` clause should resemble the following code.
```

Comment:
```suggestion
To further refine the query, you can combine multiple Boolean conditions in the `where` clause. The following code adds a condition so that the query returns those students whose first score was over 90 and whose last score was less than 80. The `where` clause should resemble the following code.
```